### DISCUSSION TITLE
Separate layers into two different solutions one for frontend and one for backend

### DISCUSSION THREAD

**Discussion by MohammdaSalah** [abpframework/abp#15560] [Q&A]
Hi, we created a multilayer MVC application using ABP template but they are placed in one visual studio solution, we want to separate them to backend solution and frontend solution so that the backend solution should only talk to the database while the frontend have the user interface logic, the goal for this is to deploy each solution in a two different servers on the production servers.
We looked at the code and the relation between the layers I'm not sure if they are cleaned separated, it does not always talk to the database from one layer, many layers are doing that.

How we can accomplish this task?

**Comment by maliming** [comment]
hi

Currently you can only modify the solution structure manually.

**Reply by MohammdaSalah** [reply]
Any ideas how to do so? ABP sometimes uses interfaces and sometimes web api to talk to backend layers. The problem is how to replace all theses interfaces to become APIs
